.eslintrc.js

```diff
@@ -1,21 +1,39 @@
 module.exports = {
   root: true,
-  extends: ['@metamask/eslint-config', '@metamask/eslint-config-nodejs'],
-  env: {
-    commonjs: true,
-  },
+
+  extends: ['@metamask/eslint-config'],
+
   overrides: [
     {
-      files: ['test/**/*.js'],
-      extends: ['@metamask/eslint-config-jest'],
+      files: ['*.ts'],
+      extends: ['@metamask/eslint-config-typescript'],
     },
-  ],
-  rules: {
-    camelcase: [
-      'error',
-      {
-        allow: ['signTypedData_v1', 'signTypedData_v3', 'signTypedData_v4'],
+
+    {
+      files: ['*.js'],
+      parserOptions: {
+        sourceType: 'script',
       },
-    ],
-  },
+      extends: ['@metamask/eslint-config-nodejs'],
+    },
+
+    {
+      files: ['*.test.ts', '*.test.js'],
+      extends: [
+        '@metamask/eslint-config-jest',
+        '@metamask/eslint-config-nodejs',
+      ],
+    },
+  ],
+
+  // This is necessary to run eslint on Windows and not get a thousand CRLF errors
+  rules: { 'prettier/prettier': ['error', { endOfLine: 'auto' }] },
```

Comment:
One more comment about this: Does it work the same if you add `endOfLine: 'auto'` to `.prettierrc.js` instead of here? Just so we configure Prettier in a consistent way.

Reply:
Yes, that also works